### PR TITLE
Feature/exam id in cookie

### PR DESCRIPTION
**Pull request by millerts**
Contains two main things:
1. Contains the necessary modules configured into the converter for the rest template to be able to parse the json from the microservices 
2. We needed to be additive in how we store data in the cookie so the legacy system can still work and we also have enough data to make the subsequent remote calls.

### CODE REVIEW

File: student.library/src/main/java/tds/student/sql/data/OpportunityInfo.java [hunk 1]
```diff
@@ -29,6 +32,11 @@ public boolean getIsOpen () {
 
   }
 
+  @JsonProperty ("IsExamOpen")
```
Comment (jtreuting) [inline student.library/src/main/java/tds/student/sql/data/OpportunityInfo.java:35]: I'm used to json properties being `isExamOpen`, is there a standard in the code now we are following?

**Review by burzum619 — APPROVED** [review]

File: student.library/src/main/java/tds/student/sql/data/OpportunityInstance.java [hunk 2]
```diff
@@ -37,9 +39,25 @@ public UUID getBrowserKey () {
     return _browserKey;
   }
 
+  @JsonProperty("ExamId")
```
Comment (burzum619) [inline student.library/src/main/java/tds/student/sql/data/OpportunityInstance.java:42]: Curious, why the Title Casing here?

Comment (millerts) [reply student.library/src/main/java/tds/student/sql/data/OpportunityInstance.java:42]: I am following the existing style.  The current code base uses Title Casing so I carried it forward in this class.

File: student/src/main/java/tds/student/web/StudentContext.java [hunk 3]
```diff
@@ -188,20 +193,24 @@ public static String getGrade () {
...
-    // check if opp instance exists
-    // if (oppKey == Guid.Empty || sessionKey == Guid.Empty || browserKey ==
-    // Guid.Empty) return null;
-    if (oppKey == null || sessionKey == null || browserKey == null)
-      return null;
+    UUID examBrowserKey = StudentCookie.getCookieData(EXAM_BROWSER_KEY) == null
+      ? null
+      : UUID.fromString(StudentCookie.getCookieData(EXAM_BROWSER_KEY));
+
+    UUID examId = tdsIdentity.get(EXAM_BROWSER_KEY) == null
```
Comment (jtreuting) [inline student/src/main/java/tds/student/web/StudentContext.java]: Is `EXAM_BROWSER_KEY` supposed to be for both of these?  `examBrowserKey` and `examId` . Maybe it should be `EXAM_ID_COOKIE_KEY` for one of them?

**Review by jtreuting — COMMENTED** [review]
Clarifying question, might need to change, not sure.

File: student.library/src/main/java/tds/student/sql/data/OpportunityInstance.java (hunk 2 above)
Comment (jtreuting) [reply student.library/src/main/java/tds/student/sql/data/OpportunityInstance.java:42]: Ooops, just saw this.  Ignore my comment on the same thing later on

File: student/src/main/java/tds/student/web/StudentContext.java (hunk 3 above)
Comment (millerts) [reply student/src/main/java/tds/student/web/StudentContext.java]: Oops.  Bit by a late refactor.  Fixed.

File: student.library/src/main/java/tds/student/sql/data/OpportunityInfo.java [hunk 4]
```diff
@@ -63,12 +71,42 @@ public void setOppKey (UUID _oppKey) {
...
+  @JsonProperty ("ExamId")
+  public UUID getExamId() {
+    return examId;
+  }
+
+  public void setExamId(UUID examId) {
+    this.examId = examId;
+  }
+
+  public void setExamStatus(OpportunityStatusType _status) {
```
Comment (jeffjohnson9046) [inline student.library/src/main/java/tds/student/sql/data/OpportunityInfo.java:92]: minor:  `ExamStatus` setter before `ExamStatus` getter.

**Review by jeffjohnson9046 — APPROVED** [review]

**Pull request merged**